### ISSUE TITLE
Transformed export statement from global exports are wrong, when re exported from module.

### ISSUE THREAD

**Issue by cebor** [rollup/rollup#1675]
The following example shows, when i re export a global export from a module the export statement in the result is wrong. 

```ts
// I would expect something like this
import * as highcharts from 'highcharts';
export { highcharts as Highcharts };

// But get this
export { Highcharts } from 'highcharts';
```

Highcharts is declared as external module and has no explicit named exports, so only the global export with the `*` works.

Example:
https://rollupjs.org/repl?version=0.51.1&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMCU3QiUyMEhpZ2hjaGFydHMlMjAlN0QlMjBmcm9tJTIwJy4lMkZpbmRleCclMjIlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIyaW5kZXguanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyaW1wb3J0JTIwKiUyMGFzJTIwSGlnaGNoYXJ0cyUyMGZyb20lMjAnaGlnaGNoYXJ0cyclNUNuZXhwb3J0JTIwJTdCJTIwSGlnaGNoYXJ0cyUyMCU3RCUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJlcyUyMiUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE

**Comment by cebor** [comment]
Issue still in v0.51.1

**Comment by lukastaegert** [comment]
Hi @cebor,

thanks for this issue and sorry you have been waiting for some time. After the easier issues caused by the recent release are settled, I plan on digging deeper into the dependency resolution algorithm, there are more issues like this one that need to be addressed.
Just be sure that you are not forgotten here!

**Comment by cebor** [comment]
Version `0.52.3` produces a new output, but even does not work properly.

Example:
https://rollupjs.org/repl?version=0.52.3&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMCU3QiUyMEhpZ2hjaGFydHMlMjAlN0QlMjBmcm9tJTIwJy4lMkZpbmRleCclMjIlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIyaW5kZXguanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyaW1wb3J0JTIwKiUyMGFzJTIwSGlnaGNoYXJ0cyUyMGZyb20lMjAnaGlnaGNoYXJ0cyclNUNuZXhwb3J0JTIwJTdCJTIwSGlnaGNoYXJ0cyUyMCU3RCUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJlcyUyMiUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE

**Comment by cebor** [comment]
Using a default export works as workaround:

```js
// main.js
export { default as Highcharts } from './index'

// indey.js
import * as Highcharts from 'highcharts'
export default Highcharts;
```

Result:
```js
import * as Highcharts from 'highcharts';



export { Highcharts };
```

Example:
https://rollupjs.org/repl?version=0.52.3&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMCU3QiUyMGRlZmF1bHQlMjBhcyUyMEhpZ2hjaGFydHMlMjAlN0QlMjBmcm9tJTIwJy4lMkZpbmRleCclMjIlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIyaW5kZXguanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyaW1wb3J0JTIwKiUyMGFzJTIwSGlnaGNoYXJ0cyUyMGZyb20lMjAnaGlnaGNoYXJ0cyclNUNuZXhwb3J0JTIwZGVmYXVsdCUyMEhpZ2hjaGFydHMlM0IlMjIlN0QlNUQlMkMlMjJvcHRpb25zJTIyJTNBJTdCJTIyZm9ybWF0JTIyJTNBJTIyZXMlMjIlMkMlMjJnbG9iYWxzJTIyJTNBJTdCJTdEJTJDJTIybmFtZSUyMiUzQSUyMm15QnVuZGxlJTIyJTJDJTIyYW1kJTIyJTNBJTdCJTIyaWQlMjIlM0ElMjIlMjIlN0QlN0QlMkMlMjJleGFtcGxlJTIyJTNBbnVsbCU3RA==

**Comment by cebor** [comment]
Version 0.55.3

This can't be correct?!

See: https://rollupjs.org/repl?version=0.55.3&shareable=JTdCJTIybW9kdWxlcyUyMiUzQSU1QiU3QiUyMm5hbWUlMjIlM0ElMjJtYWluLmpzJTIyJTJDJTIyY29kZSUyMiUzQSUyMmV4cG9ydCUyMCU3QiUyMEhpZ2hjaGFydHMlMjAlN0QlMjBmcm9tJTIwJy4lMkZpbmRleCclMjIlN0QlMkMlN0IlMjJuYW1lJTIyJTNBJTIyaW5kZXguanMlMjIlMkMlMjJjb2RlJTIyJTNBJTIyaW1wb3J0JTIwKiUyMGFzJTIwSGlnaGNoYXJ0cyUyMGZyb20lMjAnaGlnaGNoYXJ0cyclNUNuZXhwb3J0JTIwJTdCJTIwSGlnaGNoYXJ0cyUyMCU3RCUyMiU3RCU1RCUyQyUyMm9wdGlvbnMlMjIlM0ElN0IlMjJmb3JtYXQlMjIlM0ElMjJlcyUyMiUyQyUyMmdsb2JhbHMlMjIlM0ElN0IlN0QlMkMlMjJuYW1lJTIyJTNBJTIybXlCdW5kbGUlMjIlMkMlMjJhbWQlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMiUyMiU3RCU3RCUyQyUyMmV4YW1wbGUlMjIlM0FudWxsJTdE

**Comment by guybedford** [comment]
This is a special case that requires the generation of a temporary variable. We can certainly add this.

**Comment by guybedford** [comment]
(Although this would be much easier to support with the `export * as x from 'y'` proposal! - https://github.com/tc39/proposal-export-ns-from)

**Comment by guybedford** [comment]
I'm looking into reworking the code around this case as well as some other reexports scenarios. Should hopefully have a PR up soon.

**Comment by guybedford** [comment]
Should be fixed by https://github.com/rollup/rollup/pull/1947.

**Comment by guybedford** [comment]
Fixed in 0.55.4.
